src/main/java/org/seqdoop/hadoop_bam/BAMRecordReader.java

```diff
@@ -164,6 +167,11 @@ public static long getKey0(int refIdx, int alignmentStart0) {
 		readPair = false;
 		lastOfPair = false;
 		intervals = BAMInputFormat.getIntervals(conf);
+		if (intervals != null) {
+			overlapDetector = new OverlapDetector<>(0, 0);
+			overlapDetector.addAll(intervals, intervals);
```

Comment:
It might be worth [optimizing](https://github.com/samtools/htsjdk/blob/c847e260a351dd4ded3caf3c44c6ec7d0c037675/src/java/htsjdk/samtools/QueryInterval.java#L73) these intervals before handing them to the overlap detector.